api/src/main/java/org/openmrs/module/locationbasedaccess/LocationBasedAccessActivator.java

```diff
@@ -47,22 +49,33 @@ public void willStart() {
      * @see ModuleActivator#started()
      */
     public void started() {
-        String locationUserPropertyName = Context.getAdministrationService().getGlobalProperty(LocationBasedAccessConstants.REF_APP_LOCATION_USER_PROPERTY_NAME);
-        if(StringUtils.isBlank(locationUserPropertyName)) {
-            Context.getAdministrationService().setGlobalProperty(LocationBasedAccessConstants.REF_APP_LOCATION_USER_PROPERTY_NAME,
-                    LocationBasedAccessConstants.LOCATION_USER_PROPERTY_NAME);
-            if(log.isDebugEnabled()) {
-                log.debug("RefApp Location global property created with value - " + LocationBasedAccessConstants.LOCATION_USER_PROPERTY_NAME);
-            }
-        }
-        else if(StringUtils.isNotBlank(locationUserPropertyName) && !locationUserPropertyName.equals(LocationBasedAccessConstants.LOCATION_USER_PROPERTY_NAME)) {
-            log.warn("RefApp Location global property already exist in the system with different value(" + locationUserPropertyName +
-                    "). Exiting from creating new global property with the value " + LocationBasedAccessConstants.LOCATION_USER_PROPERTY_NAME);
-        }
-        log.info("Location Based Access Control Module started");
+    	if(null == Context.getPersonService().getPersonAttributeTypeByUuid(LocationBasedAccessConstants.PERSONATTRIBUTETYPE_UUID)){
+		    PersonAttributeType personAttributeType = LocationUtils.getPersonAttributeTypeForLBACModule();
+		    Context.getPersonService().savePersonAttributeType(personAttributeType);
+	    }
+	    addLBACModulesGlobalProperty(LocationBasedAccessConstants.LOCATION_ATTRIBUTE_GLOBAL_PROPERTY_NAME,
+			    LocationBasedAccessConstants.PERSONATTRIBUTETYPE_UUID);
+	    addLBACModulesGlobalProperty(LocationBasedAccessConstants.REF_APP_LOCATION_USER_PROPERTY_NAME,
+			    LocationBasedAccessConstants.LOCATION_USER_PROPERTY_NAME);
+	    log.info("Location Based Access Control Module started");
     }
 
-    /**
+	private void addLBACModulesGlobalProperty(String key, String value) {
```

Comment:
Still not giving any specific meaning in the method name! Can you reflect, the specific purpose of this method in the method name?

Reply:
createGlobalPropertyForLocationAttribute does this suit well.

Reply:
Go ahead